### PR TITLE
ppl: fix runtime issue in randomPlacement()

### PR DESCRIPTION
**Pull request by eder-matheus**
Signed-off-by: Eder Monteiro <eder.matheus.monteiro@gmail.com>

### CODE REVIEW

File: src/ppl/src/IOPlacer.cpp [hunk 1]
```diff
@@ -205,18 +204,20 @@ void IOPlacer::randomPlacement(std::vector<int> pin_indices, std::vector<int> sl
 
   std::vector<Slot> &slots = top_layer ? top_layer_slots_ : slots_;
 
+  std::vector<IOPin>& io_pins = netlist_.getIOPins();
```
Comment (maliberty) [inline src/ppl/src/IOPlacer.cpp:207]: const?

**Review by maliberty — COMMENTED** [review]
Would you explain this a bit - I'm not that familiar with this piece of code.

File: src/ppl/src/IOPlacer.cpp (hunk 1 above)
Comment (eder-matheus) [reply src/ppl/src/IOPlacer.cpp:207]: Not applicable here; I'm changing the items of this vector in the loop below.

**Comment by eder-matheus** [comment]
> Would you explain this a bit - I'm not that familiar with this piece of code.

This fix came because ioPlacer was having issues with MegaBoomX4. Some IO nets have thousands of instance pins, and this piece of code was copying these thousands of pins at every iteration of the loop.
During random pin placement, the instance pins don't matter, so I removed them from the code. It saves runtime and a lot of memory usage.

File: src/ppl/src/IOPlacer.cpp [hunk 2]
```diff
@@ -205,18 +204,20 @@ void IOPlacer::randomPlacement(std::vector<int> pin_indices, std::vector<int> sl
...
+  int io_idx = 0;
   for (int pin_idx : pin_indices) {
-    int b = vIOs[0];
+    int b = vIOs[io_idx];
```
Comment (QuantamHD) [inline src/ppl/src/IOPlacer.cpp:210]: Could we give this a more descriptive variable name?

Comment (eder-matheus) [reply src/ppl/src/IOPlacer.cpp]: Thanks for the comment! I've changed the var name.

**Review by maliberty — APPROVED** [review]

**Pull request merged**